### DISCUSSION TITLE
CSS transform translateX behaving differently when in NextJS app (Mobile)

### DISCUSSION THREAD

**Discussion by luujam2** [vercel/next.js#14006] [Help] [answered]
I've got an animation on the page which slides the new component in from the right to the center of the screen. It works fine until I switch on the chrome device emulator and select a mobile device (IPhone X). The `translateX(X%)` should take the component off the screen initially but for some reason, NextJS zooms out to fit everything on the screen causing the animation to be off. It works fine if the `translateX` value is a negative value but any positive value causes the screen to zoom out. Any help would be much appreciated!

codepen without nextJS: https://codepen.io/jl00161/full/MWKKKwy
the code for the nextJS project: https://github.com/luujam2/next-animation
link to the same example with nextJS: https://next-animation.now.sh/

**Comment by amedvedevajt** [comment, marked as answer]
`< meta name="viewport" content="width=device-width, initial-scale=1"/>` solved this problem for me

**Reply by luujam2** [reply]
thank you!

**Reply by opexchanger** [reply]
you guys have NO IDEA how much these simple question and answer have helped me! Seriously thank you a lot both of you, I've been trying to debug my animation for more than an hour lmao, as I am using framer-motion I've searched all their docs first thinking I was setting up something wrong with the library, to only later come to the conclusion the problem was the translate itself. I would probably never come to the realization myself that the viewport tag was missing, damn you nextjs for hiding it from me lmao

**Reply by sardorbek-safarov** [reply]
35hr3tg3eg

**Reply by tpudel** [reply]
Any idea for what actually causes this? Webkit stopped supporting those viewport tags back in iOS 10.
https://webkit.org/blog/7367/new-interaction-behaviors-in-ios-10/
